src/ucp/tag/tag_send.c

```diff
@@ -291,36 +332,35 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_tag_send_sync_nb,
         goto out;
     }
 
-    req = ucp_request_get(ep->worker);
-    if (req == NULL) {
-        ret = UCS_STATUS_PTR(UCS_ERR_NO_MEMORY);
-        goto out;
+    if (param->op_attr_mask & UCP_OP_ATTR_FIELD_REQUEST) {
```

Comment:
maybe define some common macro, because these lines are widely used?

Reply:
yep, will do later, when more API's will be implementing

Reply:
you already have several usages